plugins/main/public/utils/dashboards-fields.test.tsx

```diff
@@ -0,0 +1,100 @@
+
+const panelSourcesWithAgents = {
+  wellcome: '../components/common/welcome/dashboard/dashboard_panels',
+  mitre: '../components/overview/mitre/dashboard/dashboard-panels',
+  aws: '../components/overview/amazon-web-services/dashboards/dashboard_panels',
+  docker: '../components/overview/docker/dashboards/dashboard-panels',
+  fim: '../components/overview/fim/dashboard/dashboard-panels',
+  gdpr: '../components/overview/gdpr/dashboards/dashboard-panels',
+  github: '../components/overview/github/dashboards/dashboard-panels',
+  googleCloud:
+    '../components/overview/google-cloud/dashboards/dashboard_panels',
+  hipaa: '../components/overview/hipaa/dashboards/dashboard-panels',
+  malwareDetection:
+    '../components/overview/malware-detection/dashboard/dashboard-panels',
+  nist: '../components/overview/nist/dashboards/dashboard-panels',
+  pci: '../components/overview/pci/dashboards/dashboard-panels',
+  office: '../components/overview/office/dashboard/dashboard_panels',
+  threatHunting:
+    '../components/overview/threat-hunting/dashboard/dashboard_panels',
+  tsc: '../components/overview/tsc/dashboards/dashboard-panels',
+};
+
+const dashboardPanelsWithAgents = Object.fromEntries(
+  Object.entries(panelSourcesWithAgents).map(([key, path]) => [
+    key,
+    require(path).getDashboardPanels,
+  ]),
+);
+
+test('Test statistics panels in analysis engine with cluster mode', () => {
+  expect(
+    compareColumnsValue(
+      KnownFields,
+      idExtractor(getDashboardPanelsAnalysisEngine('index-pattern', true)),
+    ),
+  ).toBe(true);
+});
+
+test('Test statistics panels in analysis engine with disabled cluster mode', () => {
+  expect(
+    compareColumnsValue(
+      KnownFields,
+      idExtractor(getDashboardPanelsAnalysisEngine('index-pattern', false)),
+    ),
+  ).toBe(true);
+});
+
+test('Test statistics panels in listener engine', () => {
+  expect(
+    compareColumnsValue(
+      KnownFields,
+      idExtractor(getDashboardPanelsListenerEngine('index-pattern')),
+    ),
+  ).toBe(true);
+});
+
+Object.entries(dashboardPanelsWithAgents).forEach(([name, panelFunction]) => {
+  test(`Test ${name} panels with pinned agent`, () => {
+    expect(
+      compareColumnsValue(
+        KnownFields,
+        idExtractor(panelFunction('index-pattern', true)),
+      ),
+    ).toBe(true);
+  });
+
+  test(`Test ${name} panels without pinned agent`, () => {
+    expect(
+      compareColumnsValue(
+        KnownFields,
+        idExtractor(panelFunction('index-pattern', false)),
+      ),
+    ).toBe(true);
+  });
+});
+
+test(`Test vulnerability panels`, () => {
+  expect(
+    compareColumnsValue(
+      KnownFields,
+      idExtractor(vulnerabilitiesPanels('index-pattern')),
+    ),
+  ).toBe(true);
+});
+
+test(`Test vulnerability panels`, () => {
+  expect(
+    compareColumnsValue(
+      KnownFields,
+      idExtractor(clusterPanels('index-pattern', [], 'manager')),
```

Comment:
For the visualizations of `Cluster` you could need to extract the fields used in the `expression` field (see `q` parameter) of the visualization definition.